### PR TITLE
Added GitHub Action to build and package Docker image and upload it to DockerHub

### PR DESCRIPTION
**Pull request by DanHealey**
## Type of change
New feature

## Description
Created a GitHub Action to automatically build a Docker image from the repository, and then upload that Docker image to DockerHub, whenever a push or pull request is made to the "deployment" branch. It uses credentials from the repository's DOCKER_HUB_USERNAME and DOCKER_HUB_PASSWORD secrets to login to DockerHub and the repository's Dockerfile to build the image.

### CODE REVIEW

File: _includes/footer_custom.html [hunk 1]
```diff
@@ -1,5 +1,28 @@
 <div>
 {% if page.comments %}
-  <!---<p>Comments Here</p> --->
```
Comment (bychkov) [inline _includes/footer_custom.html:3]: @DanHealey Please move Disqus comments into the separate PR

File: _includes/nav.html [hunk 2]
```diff
@@ -64,7 +64,7 @@
         {%- if node.has_children -%}
           <a href="#" class="nav-list-expander"><svg viewBox="0 0 24 24"><use xlink:href="#svg-arrow-right"></use></svg></a>
         {%- endif -%}
-        <a href="{{ node.url | absolute_url }}" class="nav-list-link{% if page.url == node.url %} active{% endif %}">{{ node.title }}</a>
+        <a href="/snowiq-docs{{ node.url }}" class="nav-list-link{% if page.url == node.url %} active{% endif %}">{{ node.title }}</a>
```
Comment (bychkov) [inline _includes/nav.html:67]: @DanHealey  I'm pretty sure this will not work on our production website, for example https://docs.snowiq.net/soc-user-management.html

File: index.md [hunk 3]
```diff
@@ -10,18 +10,18 @@ has_toc: false
 platform: false
```
Comment (bychkov) [inline index.md:10]: @DanHealey Please make sure to merge this with the latest changes that Manolo did recently (integrating YouTube video)

**Review by bychkov — CHANGES_REQUESTED** [review]

File: _includes/nav.html (hunk 2 above)
Comment (manolo-blaufuss) [reply _includes/nav.html:67]: @DanHealey if I see this correct, this is the change you made yesterday in response to the issue we had with the relative links not being generated correctly. The links are now working again in my fork, and I think that everything should be fine without the change.

Comment (DanHealey) [reply _includes/nav.html:67]: Yes, the change was made because of the issue with relative links. I will change this.

**Comment by DanHealey** [comment]
@bychkov I implemented the needed changes, updated the repository to be up to date with manolo's, and will make a new pull request with the Disqus comments.
